### PR TITLE
3 task

### PR DESCRIPTION
**Pull request by Mavlyudabegim**
This PR is about working with databases.

I have created database in mongodb and connected it to the api with mongoose

There are models, controllers, services

### CODE REVIEW

File: server/controllers/expensesController.js [hunk 1]
```diff
@@ -1,38 +1,49 @@
...
+    return res.json(newExpense);
   } catch (error) {
-    res.send(error);
+    res.json(error);
   }
-};
+}
+async function expenses_remove(req, res) {
+  try {
+    return res.json(await expensesService.deleteOneExpence(req.params.id));
```
Comment (Lokki17) [inline server/controllers/expensesController.js:38]: Sometimes you use following syntaxis:
```
    const newExpense = await expensesService.updateExpence(
      req.params.id,
      req.body
    );
    return res.json(newExpense);
```
and sometimes:
`return res.json(await expensesService.deleteOneExpence(req.params.id));`
It's better to maintain one style

File: server/models/accounts.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const incomeSchema = require('../models/incomes');
+const mongoose = require('mongoose');
+const accountSchema = new mongoose.Schema(
+  {
+    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
+    name: { type: String, required: true, unique: true },
+    sum: { type: Number, required: true, default: 0 },
+    currencyId: { type: mongoose.Schema.Types.ObjectId, ref: 'Currency' },
+    expensesId: { type: mongoose.Schema.Types.ObjectId, ref: 'Expense' },
+    incomesId: { type: mongoose.Schema.Types.ObjectId, ref: 'Income' },
```
Comment (Lokki17) [inline server/models/accounts.js:10]: Does account has only one expense and income?

File: server/models/users.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    type: String,
+    unique: true,
+    required: [true, 'Please enter an email'],
+    lowercase: true,
+    validate: [validator.default.isEmail, 'Please enter a valid email'],
+  },
+  password: {
+    type: String,
+    required: [true, 'Please enter a password'],
+    match: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/,
```
Comment (Lokki17) [inline server/models/users.js:16]: Are you sure you need this limitation for password?

File: server/server.js [hunk 4]
```diff
@@ -1,15 +1,37 @@
+require('dotenv').config();
 const express = require('express');
 const accountRouter = require('./routes/account');
 const incomeRouter = require('./routes/income');
 const expenseRouter = require('./routes/expenses');
+const userRouter = require('./routes/user');
+const categoryRouter = require('./routes/category');
+const mongoose = require('mongoose');
+const mongoURl = `mongodb+srv://mekhrullaeva1999:Adulvam28.07@cluster0.zeoe1.mongodb.net/myFirstDatabase?retryWrites=true&w=majority`;
```
Comment (Lokki17) [inline server/server.js:9]: It's better to move mongoURI value in config

File: server/services/categoryService.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+const Category = require('../models/category');
+async function getOneCategory(id) {
+  const category = await Category.findById(id);
+  return category;
+}
+async function getAllCategories() {
+  return await Category.find();
```
Comment (Lokki17) [inline server/services/categoryService.js:7]: you dont need to use await in that case:
`return Category.find();`

**Review by Lokki17 — COMMENTED** [review]
Fix remarks

File: server/server.js (hunk 4 above)
Comment (Mavlyudabegim) [reply server/server.js:9]: My dotenv is not working. I do not know the reason , 
error: Error: ENOENT: no such file or directory, open '/home/mavlyudabegim/Documents/budgetify-1-task/.env' - this kind of error I got. 
I tried to fix this a lot. But all of them were useless.

File: server/services/categoryService.js (hunk 5 above)
Comment (Mavlyudabegim) [reply server/services/categoryService.js:7]: Can I remove all await keywords of services? Or only in getAllCategories() case

File: server/server.js (hunk 4 above)
Comment (Lokki17) [reply server/server.js:9]: Have you added .env file? Have you tried to ask google for help?

File: server/services/categoryService.js (hunk 5 above)
Comment (Lokki17) [reply server/services/categoryService.js:7]: You shouldnt add 'await' after 'return'

**Pull request merged**